### PR TITLE
[9.x] Use provided table name as foreign key if present

### PR DESCRIPTION
**Pull request by ahmett**
**TL;DR** — `getForeignKey()` method does not respect the `$table` property of model. To improve this, we use the power of `getTable()` method, also it has fallback support for the current approach ( _resolving from class basename_ ).

## Why?
 
Because, when you need to use a specific table name, you have to manually type the actual foreign key in each relation. I thought this make things more complicated, so I came up with this solution.

## What’s wrong about with tests?
 
As you can see from the tests, relation columns have named based on its models base class name, but at the same time, these models have a specific table name. As a result, column names would not match the provided table name.

To fix this, we could add the foreign keys to relations, but that's not the point of these tests, so I have decided to change these with proper convention as it should be.

## Example

```diff
class Reservation extends Model
{
  protected $table = 'flight_reservations';

  public function passengers()
  {
-    return $this->belongsToMany(Passenger::class, null, 'flight_reservation_id');
+    return $this->belongsToMany(Passenger::class);
  }
}
```

_This is probably breaking change for users who don’t follow proper naming conventions, but I believe that most people will get rid of these unnecessary definitions._

### CODE REVIEW

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.
